src/core/stubs.py

```diff
@@ -19,3 +22,27 @@
 
 
 BackupMetadata = TypedDict("BackupMetadata", {"id": str, "log-sequence-number": int, "path": str})
+
+
+class RestoreStep(str, Enum):
+    """Represent restore flow step."""
+
+    NOT_STARTED = ""
+    STOP_WORKFLOW = "stop"
+    RESTORE = "restore"
+    RESTART = "restart"
+    CLEAN = "clean"
+
+    def next_step(self) -> RestoreStep:
```

Comment:
**nitpick:** You can do `def next_step(self) -> "RestoreStep":`, and remove the `__future__` import, if you like.

Reply:
**thought:** I wonder also if a generator here would have been cool, being able to explicitly call `next()`. Not worth changing this working implementation though, but it came to mind.

Reply:
**praise:** After getting further down the review, and seeing how this was actually used, this is a really tidy solution that's very legible. MUCH easier to follow the folow than what we do for password-rotation or enabling/disabling tls, which uses the similar 'set flags on relation' pattern.